### ISSUE TITLE
Get the error: AttributeError: Can't pickle local object 'convert_frame.<locals>._convert_frame'

### ISSUE THREAD

**Issue by fladventurerob** [pytorch/torchdynamo#1967]
### 🐛 Describe the bug

When adding the line:
`model = torch.compile(model)` after loading the model, this error occurs. When removing the line, the script functions as intended. 

### Error logs

  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/opt/anaconda3/envs/ml1/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'convert_frame.<locals>._convert_frame'

**Comment by voznesenskym** [comment]
torch.compile does not emit a model in the way you expect it too, and I think maybe some new documentation has  led you astray.

If you intend to pickle the exported model, give export a try. See the export section of https://pytorch.org/get-started/pytorch-2.0/

**Comment by wconstab** [comment]
it sounds like @fladventurerob is trying to compile the model after loading it, not before exporting it.

however, @fladventurerob it would be helpful if you can provide a runnable script for us to look at, rather than just a description.

**Comment by soumith** [comment]
it might be because of multiprocess compile.
@fladventurerob can you give this a try:

1. At the top of your script, after you `import torch`, add these two lines: 

```python
from torch._inductor import config
config.compile_threads = 1
```

2. Then, add back the `torch.compile` call.

Does the error reproduce?

**Comment by fladventurerob** [comment]
> ```python
> from torch._inductor import config
> config.compile_threads = 1
> ```

You are correct. The model exists already. I was loading it into a forward test script. Based upon the documentation I was assuming this line was needed to use an existing model, rather than to export the model.

**Comment by fladventurerob** [comment]
> it sounds like @fladventurerob is trying to compile the model after loading it, not before exporting it.
> 
> however, @fladventurerob it would be helpful if you can provide a runnable script for us to look at, rather than just a description.

You are correct. The model exists already. I was loading it into a forward test script. Based upon the documentation I was assuming this line was needed to use an existing model, rather than to export the model.

**Comment by wconstab** [comment]
@fladventurerob any update on whether compile_threads=1 helps, or are you able to provide a repro script for us?

**Comment by ConnorBaker** [comment]
I had the same issue -- setting `compile_threads=1` fixed it for me!
